sdk/eventgrid/azure-eventgrid/README.md

```diff
@@ -24,31 +36,47 @@ Install the Azure Event Grid client library for Python with [pip][pip]:
 pip install azure-eventgrid
 ```
 
-* An existing Event Grid topic or domain is required. You can create the resource using [Azure Portal][azure_portal_create_EG_resource] or [Azure CLI][azure_cli_link]
+* An existing Event Grid Basic topic or domain, or Event Grid Namespace topic is required. You can create the resource using [Azure Portal][azure_portal_create_EG_resource] or [Azure CLI][azure_cli_link]
 
 If you use Azure CLI, replace `<resource-group-name>` and `<resource-name>` with your own unique names.
 
-#### Create an Event Grid Topic
+#### Create an Event Grid Namespace
 
 ```
-az eventgrid topic --create --location <location> --resource-group <resource-group-name> --name <resource-name>
+az eventgrid namespace create --location <location> --resource-group <resource-group-name> --name <resource-name>
 ```
 
-#### Create an Event Grid Domain
+#### Create an Event Grid Namespace Topic
 
 ```
-az eventgrid domain --create --location <location> --resource-group <resource-group-name> --name <resource-name>
+az eventgrid namespace create topic --location <location> --resource-group <resource-group-name> --name <resource-name>
 ```
 
 ### Authenticate the client
 In order to interact with the Event Grid service, you will need to create an instance of a client.
 An **endpoint** and **credential** are necessary to instantiate the client object.
 
+
+The default EventGridPublisherClient created is compatible with an Event Grid Basic Resource. To create an Event Grid Namespace compatible Client, specify `namespace_topic="YOUR_TOPIC_NAME"` when instantiating the client.
+
+```python
+# Event Grid Namespace client
+client = EventGridPublisherClient(endpoint, credential, namespace_topic=YOUR_TOPIC_NAME)
+```
+
+EventGridConsumerClient only supports EventGrid Namespaces.
+```python
+# Event Grid Namespace Client
+client = EventGridConsumerClient(endpoint, credential, namespace_topic=YOUR_TOPIC_NAME, subscription=YOUR_SUBSCRIPTION_NAME)
+```
+
 #### Using Azure Active Directory (AAD)
 
 Azure Event Grid provides integration with Azure Active Directory (Azure AD) for identity-based authentication of requests. With Azure AD, you can use role-based access control (RBAC) to grant access to your Azure Event Grid resources to users, groups, or applications.
 
 To send events to a topic or domain with a `TokenCredential`, the authenticated identity should have the "EventGrid Data Sender" role assigned.
+To receive events to a topic with a `TokenCredential`, the authenticated identity should have the "EventGrid Data Receiver" role assigned.
+To send and receive events to/from a topic with a `TokenCredential`, the authenticated identity should have the "EventGrid Data Contributor" role assigned.
```

Comment:
might be helpful to link docs on how to assign rbac: https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-steps